src/test/java/part2/exercise/CollectorsExercise1.java

```diff
@@ -110,12 +139,114 @@ public void getTheCoolestOne2() {
                                 ppd -> ppd.get().getPerson())));
     }
 
-    private Map<String, Person> getCoolestByPosition2WithCustomCollector(List<Employee> employees) {
-        throw new UnsupportedOperationException(); // TODO: написать свой коллектор
-        // With the longest sum duration on this position
+    private static class JobHistoryCombiner implements Collector<JobHistoryEntry, Map<String, int[]>, Stream<PersonPositionDuration>> {

```

Comment:
int[] is faster, especially when mutating map value for a given key. Standard implementation of summingInt collector uses int[1] internally.